File: 0867. Transpose Matrix/0867.java
// time complexity: O(m*n)
// space complexity: O(m*n)
class Solution {
    public int[][] transpose(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        int[][] transposed = new int[n][m];
        for(int i=0; i<matrix.length; i++){
            for(int j=0; j<matrix[i].length; j++){
                transposed[j][i] = matrix[i][j];
            }
        }
        return transposed;
    }
}

File: 0867. Transpose Matrix/0867.py
# time complexity: O(m*n)
# space compleixty: O(m*n)
class Solution(object):
    def transpose(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[List[int]]
        """
        m, n = len(matrix), len(matrix[0])
        # return the transposed matrix
        return [[matrix[i][j] for i in range(m)] for j in range(n)]